AppCenterData/AppCenterData/Internal/MSDocumentWrapperInternal.h

```diff
@@ -27,18 +26,55 @@
                                      eTag:(NSString *)eTag
                           lastUpdatedDate:(NSDate *)lastUpdatedDate
                          pendingOperation:(NSString *)pendingOperation
-                                    error:(MSDataError *)error
                           fromDeviceCache:(BOOL)fromDeviceCache;
 
+/**
+ * Initialize a `MSDocumentWrapper` instance.
+ *
+ * @param deserializedValue The document value. Must conform to MSSerializableDocument protocol.
+ * @param jsonValue The document's JSON representation.
+ * @param partition Partition key.
+ * @param documentId Document id.
+ * @param eTag Document eTag.
+ * @param lastUpdatedDate Last updated date of the document.
+ * @param pendingOperation The name of the pending operation, or nil.
+ * @param fromDeviceCache Flag indicating if the document wrapper was retrieved remotely or not.
+ * @param error Document error.
+ *
+ * @return A new `MSDocumentWrapper` instance.
+ */
+- (instancetype)initWithDeserializedValue:(id<MSSerializableDocument>)deserializedValue
+                                jsonValue:(NSString *)jsonValue
+                                partition:(NSString *)partition
+                               documentId:(NSString *)documentId
+                                     eTag:(NSString *)eTag
+                          lastUpdatedDate:(NSDate *)lastUpdatedDate
+                         pendingOperation:(NSString *)pendingOperation
+                          fromDeviceCache:(BOOL)fromDeviceCache
+                                    error:(MSDataError *)error;
+
 /**
  * Initialize a `MSDocumentWrapper` instance.
  *
  * @param error Document error.
  * @param documentId Document Id.
+ * @param partition Partition key.
+ *
+ * @return A new `MSDocumentWrapper` instance.
+ */
+- (instancetype)initWithError:(MSDataError *)error documentId:(NSString *)documentId partition:(NSString *)partition;
```

Comment:
Do we need this initializer when the one below exists? Can just pass `nil` for eTag if it doesn't exist or isn't known.

Reply:
@achocron is this just a convention that we prefer is to pass `nil` instead of overloading methods?

Reply:
@achocron, we add this method overload to avoid changing everywhere that invokes this initializer. This initializer is called in 10+ places in SDK (excluding tests) but the below initializer is called in only 1 place. 

Any other reason against adding this method overload?